src/main/java/wolox/training/models/Book.java

```diff
@@ -142,6 +151,7 @@ public String getIsbn() {
     }
 
     public void setIsbn(String isbn) {
+        Preconditions.checkNotNull(isbn);
```

Comment:
After checking if a field is null, check if the field is empty, and then check if it is numeric.

An error message can be added to each check :

```
Preconditions.checkNotNull(isbn, "The isbn field cannot be null.");
Preconditions.checkArgument(! isbn.isEmpty(), "The isbn field cannot be empty.");
Preconditions.checkArgument(StringUtils.isNumeric(isbn), "error message");
this. isbn = isbn;
```

I recommend using a class to store error messages as constants.

Reply:
I put message in constant and added validation nueric